### DISCUSSION TITLE
How to build Metrics using OpenTelemetry SDK?

### DISCUSSION THREAD

**Discussion by swapnil-kotwal-sp** [open-telemetry/community#1378] [Community Discussions]
Hey Team,

I've just started my journey with OpenTelemetry. My basic Objective is to get the Metrics for no. of successful & failed operations.
I've written below code sniped, I could see my data is getting ingested to Prometheus but 

Que 1: does below piece of code is correct as per my expectations?
Que 2: I'm not getting any details about span attributes or tags etc... What I'm missing here? When I decompiled this code.. There I could see span section as `Exception exception = null`
Que 3: I'm adding successful `counter.add(1);` same way I will have to instantiate -ve counter object for unsuccessful operations?

``` 
		OpenTelemetry openTelemetry = ExampleConfiguration.initOpenTelemetry();
		Tracer tracer = openTelemetry.getTracer("test instrumentation", "1.0.0");
		// Gets or creates a named meter instance
		Meter meter = openTelemetry.meterBuilder("test instrumentation")
				.setInstrumentationVersion("1.0.0").build();
		// Build counter e.g. LongCounter
		LongCounter counter = meter
				.counterBuilder("test_counter")
				.setDescription("test_counter jobs")
				.setUnit("1")
				.build();
		LongHistogram histogram = meter.histogramBuilder("super_timer").ofLongs().setUnit("ms").build();

		for (int i = 0; i < 100; i++) {
			long startTime = System.currentTimeMillis();
			Span span = tracer.spanBuilder("exampleSpan").startSpan();

			try (Scope scope = span.makeCurrent()) {
				span.setAttribute("good", "true");
				span.setAttribute("exampleNumber", i);
				counter.add(1);
				span.setStatus(StatusCode.OK, "Something good happened!");
				Thread.sleep(100);
			} catch (Throwable throwable) {
				//counter.add(1); shall I add -ve counter here?
				span.setStatus(StatusCode.ERROR, "Something bad happened!");
				span.recordException(throwable);
				throw throwable;
			} finally {
				histogram.record(System.currentTimeMillis() - startTime);
				span.end();
			}
		}
```
<img width="1682" alt="Screenshot 2023-03-07 at 9 34 53 AM" src="https://user-images.githubusercontent.com/1849261/223318083-eafccaa7-34b7-458f-874b-82a3f74d5a51.png">

**Comment by trask** [comment]
hi @swapnil-kotwal-sp! I'd suggest opening discussion in https://github.com/open-telemetry/opentelemetry-java/discussions

**Reply by swapnil-kotwal-sp** [reply]
Thank You, posted here https://github.com/open-telemetry/opentelemetry-java/discussions/5274
